src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1116,4 +1117,48 @@ function ($a) {
             ->end()
         ;
     }
+
+    private function addEncryptedSecretsSection(ArrayNodeDefinition $rootNode)
+    {
+        $rootNode
+            ->children()
+                ->arrayNode('encrypted_secrets')
+                    ->info('encrypted secrets configuration')
+                    ->canBeEnabled()
+                    ->children()
+                        ->scalarNode('secrets_file')
+                            ->info('Set the file location to read encrypted secrets from')
+                            ->example("'%kernel.project_dir%/config/secrets.enc.json' to set the file location to config/secrets.enc.json")
+                            ->defaultValue('%kernel.project_dir%/config/secrets.enc.json')
+                            ->treatNullLike(array('enabled' => false))
+                            ->validate()
+                                ->ifTrue(function ($v) { return !(\is_string($v) || \is_null($v)); })
+                                ->thenInvalid('If set, the "encrypted_secrets.secrets_file" should be a string path to an existing secrets file')
+                            ->end()
+                        ->end()
+                        ->scalarNode('private_key_file')
+                            ->info('Set the file location to read the private decryption key from')
+                            ->example("/etc/your-project-name/master-key")
+                            ->defaultNull()
+                            ->treatNullLike(array('enabled' => false))
+                            ->validate()
+                                ->ifTrue(function ($v) { return !(\is_string($v) || \is_null($v)); })
+                                ->thenInvalid('If set, the "encrypted_secrets.private_key_file" should be a string path to an existing file containing a 32 byte private key')
+                            ->end()
+                        ->end()
+                        ->scalarNode('public_key_file')
+                            ->info('Set the file location to read the public encryption key from')
+                            ->example('%kernel.project_dir%/config/secrets.pub')
+                            ->defaultNull()
+                            ->treatNullLike(array('enabled' => false))
+                            ->validate()
+                                ->ifTrue(function ($v) { return !(\is_string($v) || \is_null($v)); })
+                                ->thenInvalid('If set, the "encrypted_secrets.public_key_file" should be a string path to an existing file containing a 32 byte public key')
+                            ->end()
+                        ->end()
```

Comment:
Can we make both `private_key_file` and `public_key_file` required once they have enabled the secrets feature?

Reply:
yup!

Reply:
This is (partially) addressed in the latest commit, I have a few questions though:

 If I make private_key_file required, this will mess with the developer flow for being able to add secrets with only a public key. Currently, decrypting/editing secrets is only allowed for setups with both the public and private key, but with the public key in version control, developers can run `secrets:add` and add or overwrite secrets without ever needing the private key.

For now I added the validation checks for string presence on secrets_file and public_key_file if enabled is true. (Disclaimer: I also provided default values, so this is only stopping them from setting it to a null, integer, etc. value manually). Thoughts?